### ISSUE TITLE
Cannot upgrade ruby on macos due to gcc version mismatch

### ISSUE THREAD

**Issue by crowell** [rvm/rvm#4546]
## Description

i have ruby 2.4.1 installed, trying to install ruby 2.6.0, install requirements say i need gcc7, but i have gcc8 installed, and the gcc7 keg is not available in homebrew any more.

```
rvm upgrade 2.4.1 2.6
Are you sure you wish to upgrade from ruby-2.4.1 to ruby-2.6.0? (Y/n): Y
Installing new ruby ruby-2.6.0
Searching for binary rubies, this might take some time.
No binary rubies available for: osx/10.14/x86_64/ruby-2.6.0.
Continuing with compilation. Please read 'rvm help mount' to get more information on binary rubies.
Checking requirements for osx.
Error: No such keg: /usr/local/Cellar/gcc@7
Requirements installation failed with status: 1.
Unable to install ruby ruby-2.6.0. Please install it manually to continue.
```

## Steps to reproduce

```
rvm upgrade 2.4.1 2.6
Are you sure you wish to upgrade from ruby-2.4.1 to ruby-2.6.0? (Y/n): Y
Installing new ruby ruby-2.6.0
Searching for binary rubies, this might take some time.
No binary rubies available for: osx/10.14/x86_64/ruby-2.6.0.
Continuing with compilation. Please read 'rvm help mount' to get more information on binary rubies.
Checking requirements for osx.
Error: No such keg: /usr/local/Cellar/gcc@7
Requirements installation failed with status: 1.
Unable to install ruby ruby-2.6.0. Please install it manually to continue.
```

## Expected behavior

ruby 2.6.0 is installed

## Actual behavior

* [command](https://gist.github.com/...)
* [config.log](https://gist.github.com/...) - only for issues with `install`
* [make.log](https://gist.github.com/...) - only for issues with `install`

```
rvm upgrade 2.4.1 2.6
Are you sure you wish to upgrade from ruby-2.4.1 to ruby-2.6.0? (Y/n): Y
Installing new ruby ruby-2.6.0
Searching for binary rubies, this might take some time.
No binary rubies available for: osx/10.14/x86_64/ruby-2.6.0.
Continuing with compilation. Please read 'rvm help mount' to get more information on binary rubies.
Checking requirements for osx.
Error: No such keg: /usr/local/Cellar/gcc@7
Requirements installation failed with status: 1.
Unable to install ruby ruby-2.6.0. Please install it manually to continue.
```

**Debug mode**:
* [command](https://gist.github.com/...)
* [config.log](https://gist.github.com/...) - only for issues with `install`
* [make.log](https://gist.github.com/...) - only for issues with `install`
```
rvm upgrade 2.4.1 2.6 --debug
Are you sure you wish to upgrade from ruby-2.4.1 to ruby-2.6.0? (Y/n): y
Installing new ruby ruby-2.6.0
ruby-2.6.0 - install
ruby-2.6.0 - #already removed src/ruby-2.6.0
ruby-2.6.0 - #already removed rubies/ruby-2.6.0
Searching for binary rubies, this might take some time.
Running(18): curl --fail --location --max-redirs 10 --max-time 1800 --silent --insecure --location --list-only --max-time 5 --head -X GET https://rvm_io.global.ssl.fastly.net/binaries/osx/10.14/x86_64/ruby-2.6.0.tar.bz2 --insecure
Remote file does not exist https://rvm_io.global.ssl.fastly.net/binaries/osx/10.14/x86_64/ruby-2.6.0.tar.bz2
Running(18): curl --fail --location --max-redirs 10 --max-time 1800 --silent --insecure --location --list-only --max-time 5 --head -X GET https://github.com/oracle/truffleruby/releases/download/ruby-2.6.0.tar.bz2 --insecure
Remote file does not exist https://github.com/oracle/truffleruby/releases/download/ruby-2.6.0.tar.bz2
Running(18): curl --fail --location --max-redirs 10 --max-time 1800 --silent --insecure --location --list-only --max-time 5 --head -X GET https://repo1.maven.org/maven2/org/jruby/jruby-dist/ruby-2.6.0.tar.bz2 --insecure
Remote file does not exist https://repo1.maven.org/maven2/org/jruby/jruby-dist/ruby-2.6.0.tar.bz2
Travis binaries for OSX are not movable and can't be used outside of Travis environment. Skip that source.
No remote file name found
Running(18): curl --fail --location --max-redirs 10 --max-time 1800 --silent --insecure --location --list-only --max-time 5 --head -X GET https://rvm_io.global.ssl.fastly.net/binaries/osx/10.14/x86_64/ruby-2.6.0.tar.bz2 --insecure
Remote file does not exist https://rvm_io.global.ssl.fastly.net/binaries/osx/10.14/x86_64/ruby-2.6.0.tar.bz2
Running(18): curl --fail --location --max-redirs 10 --max-time 1800 --silent --insecure --location --list-only --max-time 5 --head -X GET https://github.com/oracle/truffleruby/releases/download/ruby-2.6.0.tar.bz2 --insecure
Remote file does not exist https://github.com/oracle/truffleruby/releases/download/ruby-2.6.0.tar.bz2
Running(18): curl --fail --location --max-redirs 10 --max-time 1800 --silent --insecure --location --list-only --max-time 5 --head -X GET https://repo1.maven.org/maven2/org/jruby/jruby-dist/ruby-2.6.0.tar.bz2 --insecure
Remote file does not exist https://repo1.maven.org/maven2/org/jruby/jruby-dist/ruby-2.6.0.tar.bz2
Travis binaries for OSX are not movable and can't be used outside of Travis environment. Skip that source.
No remote file name found
No binary rubies available for: osx/10.14/x86_64/ruby-2.6.0.
Continuing with compilation. Please read 'rvm help mount' to get more information on binary rubies.
Free disk space 188189MB, required 440MB.
__rvm_setup_compile_environment_setup ruby-2.6.0
rvm_autolibs_flag=4
__rvm_setup_compile_environment_movable_early ruby-2.6.0
__rvm_setup_compile_environment_system_early ruby-2.6.0
__rvm_setup_compile_environment_requirements ruby-2.6.0
Checking requirements for osx.
requirements code for osx loaded
requirements lib type set to osx_brew
brew seems to be writable
Error: No such keg: /usr/local/Cellar/gcc@7
Requirements installation failed with status: 1.
Unable to install ruby ruby-2.6.0. Please install it manually to continue.
__rvm_rm_rf already gone: /Users/jeff/.rvm/tmp/32390*
```

## Environment info

```
ruby-2.4.1:

  system:
    uname:        "Darwin busytown.local 18.2.0 Darwin Kernel Version 18.2.0: Mon Nov 12 20:24:46 PST 2018; root:xnu-4903.231.4~2/RELEASE_X86_64 x86_64"
    name:         "OSX"
    version:      "10.14"
    architecture: "x86_64"
    bash:         "/bin/bash => GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin18)"
    zsh:          "/usr/local/bin/zsh => zsh 5.6.2 (x86_64-apple-darwin18.0.0)"
    remote_path:  "osx/10.14/x86_64"
    xcode:        "10.1"

  rvm:
    version:      "1.29.7 (master)"
    updated:      "2 minutes 3 seconds ago"
    path:         "/Users/jeff/.rvm"
    autolibs:     "[4] Allow RVM to use package manager if found, install missing dependencies, install package manager (only OS X)."

  ruby:
    interpreter:  "ruby"
    version:      "2.4.1p111"
    date:         "2017-03-22"
    platform:     "x86_64-darwin17"
    patchlevel:   "2017-03-22 revision 58053"
    full_version: "ruby 2.4.1p111 (2017-03-22 revision 58053) [x86_64-darwin17]"

  homes:
    gem:          "/Users/jeff/.rvm/gems/ruby-2.4.1"
    ruby:         "/Users/jeff/.rvm/rubies/ruby-2.4.1"

  binaries:
    ruby:         "/Users/jeff/.rvm/rubies/ruby-2.4.1/bin/ruby"
    irb:          "/Users/jeff/.rvm/rubies/ruby-2.4.1/bin/irb"
    gem:          "/Users/jeff/.rvm/rubies/ruby-2.4.1/bin/gem"
    rake:         "/Users/jeff/.rvm/rubies/ruby-2.4.1/bin/rake"

  environment:
    PATH:         "/Users/jeff/.rvm/gems/ruby-2.4.1/bin:/Users/jeff/.rvm/gems/ruby-2.4.1@global/bin:/Users/jeff/.rvm/rubies/ruby-2.4.1/bin:/usr/local/opt/libarchive/bin:/usr/local/sbin:/usr/local/opt/qt/bin:/Users/jeff/.rvm/bin:/Users/jeff/.cargo/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Applications/VMware Fusion.app/Contents/Public/:/opt/X11/bin:/Library/Frameworks/Mono.framework/Versions/Current/Commands:/Applications/Wireshark.app/Contents/MacOS:/Users/jeff/.rvm/gems/ruby-2.4.1/bin:/Users/jeff/.rvm/gems/ruby-2.4.1@global/bin:/Users/jeff/.rvm/rubies/ruby-2.4.1/bin:/usr/local/opt/libarchive/bin:/usr/local/sbin:/usr/local/opt/qt/bin:/Users/jeff/.rvm/bin:/Users/jeff/.cargo/bin:/Users/jeff/dev/bin:/Users/jeff/.config/radare2/prefix/bin:/Users/jeff/.rvm/bin:/Users/jeff/dev/bin:/Users/jeff/.config/radare2/prefix/bin"
    GEM_HOME:     "/Users/jeff/.rvm/gems/ruby-2.4.1"
    GEM_PATH:     "/Users/jeff/.rvm/gems/ruby-2.4.1:/Users/jeff/.rvm/gems/ruby-2.4.1@global"
    MY_RUBY_HOME: "/Users/jeff/.rvm/rubies/ruby-2.4.1"
    IRBRC:        "/Users/jeff/.rvm/rubies/ruby-2.4.1/.irbrc"
    RUBYOPT:      ""
    gemset:       ""
```

**Comment by crowell** [comment]
this diff may work

```diff
diff --git a/scripts/functions/requirements/osx_brew b/scripts/functions/requirements/osx_brew
index f80c6a11..35a3d72b 100644
--- a/scripts/functions/requirements/osx_brew
+++ b/scripts/functions/requirements/osx_brew
@@ -208,9 +208,10 @@ run 'brew doctor' and make sure latest '$brew_openssl_package' is installed prop
 requirements_osx_brew_libs_default_check_gcc()
 {
   if
-    requirements_detect_installed gcc@7 gcc@6 gcc@5 gcc@4.9 gcc48 gcc47 gcc46 apple-gcc42
+    requirements_detect_installed gcc@8 gcc@7 gcc@6 gcc@5 gcc@4.9 gcc48 gcc47 gcc46 apple-gcc42
   then
     case "${packages_installed[*]}" in
+      (*gcc@8*)       export CC="$(brew --prefix gcc@8      )/bin/gcc-8" ;;
       (*gcc@7*)       export CC="$(brew --prefix gcc@7      )/bin/gcc-7" ;;
       (*gcc@6*)       export CC="$(brew --prefix gcc@6      )/bin/gcc-6" ;;
       (*gcc@5*)       export CC="$(brew --prefix gcc@5      )/bin/gcc-5" ;;
@@ -247,6 +248,7 @@ requirements_osx_brew_libs_default_add_gcc_v()
     (5.0) requirements_osx_brew_libs_default_add_gcc "$2" gcc-5.0 gcc@5       ;;
     (6.0) requirements_osx_brew_libs_default_add_gcc "$2" gcc-6.0 gcc@6       ;;
     (7.0) requirements_osx_brew_libs_default_add_gcc "$2" gcc-7.0 gcc@7       ;;
+    (8.0) requirements_osx_brew_libs_default_add_gcc "$2" gcc-8.0 gcc@8       ;;
     (*)
       rvm_error "Do not know how to check/install gcc '$1'."
       return 1
```

**Comment by pkuczynski** [comment]
This should be fixed with #4557 and #4556. Please do `rvm get master` and try again. Let us know in case of the problem still existing.

**Comment by pkuczynski** [comment]
We just released a new version 1.29.7 including this fix.

[Become a backer](https://opencollective.com/rvm#backer) and support us with a small donation to help us continue our activities. Thank you if you already one! 🙏
